### ISSUE TITLE
Bug: Control Server is broken for Wireguard (works only for OpenVPN)

### ISSUE THREAD

**Issue by Aubermean** [qdm12/gluetun#2438]
### Is this urgent?

Yes

### CPU arch

aarch64

### VPN service provider

Custom

### What are you using to run the container

docker-compose

### What is the version of Gluetun

 2024-03-07T12:32:25.391Z (commit 3254fc8)

### What's the problem 🤔

Using the Control Server, trying to 'start' a wireguard server results in some bug whereby it switches over to openvpn and the container needs restarting.

This version works:  2024-03-07T12:32:25.391Z (commit 3254fc8)

the latest, v3.39, v3.38, v3.37 tags do not work.

I don't use OpenVPN, I never have. The commit tag above works fine, this log I am adding is from the latest versions so you can see it decides to switch to openvpn after i stop-start via control server.

### Share your logs (at least 10 lines)

```plain text
gluetun_a-1  | 2024-08-24T20:22:38Z INFO [wireguard] Using available kernelspace implementation
gluetun_a-1  | 2024-08-24T20:22:38Z INFO [wireguard] Connecting to ....
gluetun_a-1  | 2024-08-24T20:22:38Z INFO [wireguard] Wireguard setup is complete. Note Wireguard is a silent protocol and it may or may not work, without giving any error message. Typically i/o timeout errors indicate the Wireguard connection is not working.
gluetun_a-1  | 2024-08-24T20:22:38Z INFO [dns] downloading DNS over TLS cryptographic files
gluetun_a-1  | 2024-08-24T20:22:39Z INFO [healthcheck] healthy!
gluetun_a-1  | 2024-08-24T20:22:40Z INFO [dns] downloading hostnames and IP block lists
gluetun_a-1  | 2024-08-24T20:22:45Z INFO [dns] init module 0: validator
gluetun_a-1  | 2024-08-24T20:22:45Z INFO [dns] init module 1: iterator
gluetun_a-1  | 2024-08-24T20:22:45Z INFO [dns] start of service (unbound 1.20.0).
gluetun_a-1  | 2024-08-24T20:22:45Z INFO [dns] generate keytag query ....
gluetun_a-1  | 2024-08-24T20:22:45Z INFO [dns] ready
gluetun_a-1  | 2024-08-24T20:22:46Z INFO [ip getter] Public IP address is ....
gluetun_a-1  | 2024-08-24T20:22:46Z INFO [vpn] There is a new release v3.39.0 (v3.39.0) created 15 days ago
gluetun_a-1  | 2024-08-24T20:22:48Z INFO [http server] 200 GET /ip wrote 210B to 172.16.238.1:54854 in 182.84µs
gluetun_a-1  | 2024-08-24T20:22:53Z INFO [vpn] stopping
gluetun_a-1  | 2024-08-24T20:22:53Z INFO [vpn] starting
gluetun_a-1  | 2024-08-24T20:22:53Z INFO [firewall] allowing VPN connection...
gluetun_a-1  | 2024-08-24T20:22:53Z ERROR [openvpn] No client-side authentication method is specified.  You must use either --cert/--key, --pkcs12, or --auth-user-pass
gluetun_a-1  | 2024-08-24T20:22:53Z INFO [openvpn] Use --help for more information.
gluetun_a-1  | 2024-08-24T20:22:53Z ERROR [vpn] exit status 1
gluetun_a-1  | 2024-08-24T20:22:53Z INFO [vpn] retrying in 15s
```

### Share your configuration

```yml
no
```

**Comment by qdm12** [comment]
Which route are you using to (re)start wireguard through the control server?
Also can you share some logs, for example in the latest image to see how it misbehaves? Thanks!

**Comment by Aubermean** [comment]
Hi Quentin, latest behaves the same as log above, but attaching full log using latest for your reference. `v1` is generally quite an (ahem) 'anti-pattern' API, but also very useful! `v2` one day would be great. `<3`

```bash
> curl -X PUT http://localhost:8005/v1/openvpn/status -H "Content-Type: application/json" -d '{"status": "stopped"}'
{"outcome":"stopped"}
> curl -X PUT http://localhost:8005/v1/openvpn/status -H "Content-Type: application/json" -d '{"status": "running"}'
{"outcome":"crashed"}
```
[gluetun_log.txt](https://github.com/user-attachments/files/16739657/gluetun_log.txt)

**Comment by Aubermean** [comment]
Hi, so the endpoint was renamed from `/v1/openvpn` to `/v1/vpn`?

Since you made the wise choice of starting off with api versioning, if you change the API it would really make sense to use `v2` and leave the old endpoints so things don't break. Otherwise I have to integrate testing on my side to make sure new releases don't break the API client library I have written.

Also #2238 doesn't make sense I think there is a typo?

**Comment by Aubermean** [comment]
> In spirit of easier maintainability, asking for backwards compatibility for undocumented api is a bit much don't you think? @Aubermean I think you should have a test suite on your end to make sure your needs are served well.

You are mistaken, the API has never been [undocumented](https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/control-server.md). The API /v1/ had its endpoints changed completely and silently and the documentation was not updated.

Speaking of easier maintainabilitiy, the solution is definitely not everyone having 'a test suite on their end'. The solution is:

- Making use of the API versioning that is already implemented
- Updating the documentation should the API change
- Avoiding anti-patterns to follow good design principles with all iterative changes

**Comment by qdm12** [comment]
A bit late to the party, but...

@Aubermean I would agree with your take on not breaking compatibility; however this compatibility break you are mentioning is utter hallucination. The control server always maintained compatibility. Literally still works today with the /openvpn path.

https://github.com/qdm12/gluetun/blob/f6d44742876180be1736411a1a0149ecc9427b94/internal/server/openvpn.go#L35

https://github.com/qdm12/gluetun/blob/f6d44742876180be1736411a1a0149ecc9427b94/internal/server/openvpn.go#L42

> There are routes defined in the code but not documented

I meant the `/vpn/*` routes back then, that's the ones that were still undocumented.

> Once it's documented it will stay retro-compatible.

Literally said that, why would you think against me saying this!?

...

And for the sake of nailing a thick nail in this coffin of accusations, even going back 2024-08-25 matching d82df2b431f96636bbd11367054d6f73033a6f34 see for yourself:

https://github.com/qdm12/gluetun/blob/d82df2b431f96636bbd11367054d6f73033a6f34/internal/server/openvpn.go#L37

https://github.com/qdm12/gluetun/blob/d82df2b431f96636bbd11367054d6f73033a6f34/internal/server/openvpn.go#L46

Lessons to be learnt:

1. Double check before making accusations
2. Read more slowly
